control-plane/agents/src/bin/ha/cluster/switchover.rs

```diff
@@ -243,48 +247,55 @@ impl SwitchOverEngine {
             nodes,
         };
 
-        sw.mover(rq_rx);
+        sw.mover(Arc::new(Mutex::new(rq_rx)));
         sw
     }
 
-    /// Mover is responsible for processing the switchover request sequentially.
-    pub fn mover(&self, mut recv: UnboundedReceiver<SwitchOverRequest>) {
+    /// Switchover request to be handled synchronously in each worker task.
+    async fn process_request(&self, recv: Arc<Mutex<UnboundedReceiver<SwitchOverRequest>>>) {
         let etcd = self.etcd.clone();
         let nodes = self.nodes.clone();
-
-        tokio::spawn(async move {
-            loop {
-                if let Some(mut q) = recv.recv().await {
-                    let retry_delay_min = std::time::Duration::from_secs(1);
-                    let retry_delay_max = std::time::Duration::from_secs(384);
-                    let mut retry_delay = retry_delay_min;
-                    loop {
-                        // TODO: handle error handling properly
-                        let result = match q.stage {
-                            Stage::Init => q.initialize(&etcd).await,
-                            Stage::RepublishVolume => q.republish_volume(&etcd).await,
-                            Stage::PublishPath => q.publish_path(&etcd, &nodes).await,
-                            Stage::DeleteTarget => q.delete_target(&etcd).await,
-                            Stage::Successful => match q.delete_switchover(&etcd).await {
-                                Ok(_) => break,
-                                Err(e) => Err(e),
-                            },
-                            _ => break,
-                        };
-                        match result {
-                            Ok(_) => {
-                                // reset retry delay back to the start.
-                                retry_delay = retry_delay_min;
-                            }
-                            Err(_) => {
-                                retry_delay = (retry_delay * 3).min(retry_delay_max);
-                                tokio::time::sleep(retry_delay).await;
-                            }
+        loop {
```

Comment:
You don't need to clone etcd and nodes now?

Reply:
Incorporated